pp_hot.c

```diff
@@ -5995,8 +5995,10 @@ Perl_leave_adjust_stacks(pTHX_ SV **from_sp, SV **to_sp, U8 gimme, int pass)
                  * SvTYPE(sv), where that is a SVt_PVNV or below. It is
                  * more efficient to create such types directly than
                  * upgrade to them via sv_upgrade() within sv_setsv_flags. */
-                SV *newsv = (SvTYPE(sv) <= SVt_PVNV)
-                            ? newSV_type(SvTYPE(sv))
+                SV *newsv = SvTYPE(sv) == SVt_IV ? newSV_type(SVt_IV)
+#if NVSIZE <= IVSIZE
+                            : SvTYPE(sv) == SVt_NV ? newSV_type(SVt_NV)
+#endif
```

Comment:
I haven't had time to check, but is this really correct? I thought `SvTYPE(sv)` could be types greater than `SVt_NV` at this point.

Reply:
those are standard boiler plate handling of body-less NV optimization that I added years ago. IV all CPUs and OSes and NV on AMD x64 are bodyless so they have a much faster escape path in that function, vs  than full body types.